### DISCUSSION TITLE
Add `stateOverrides` to `estimateGas`

### DISCUSSION THREAD

**Discussion by conwayconstar** [wevm/viem#2234] [Idea / Feature Request] [answered]
`stateOverrides` were added to the `eth_estimateGas` request in [geth in August 2023 ](https://github.com/ethereum/go-ethereum/pull/27845) was wondering if it could be implmented into Viems `estimateGas` action

**Comment by roninjin10** [comment]
This would help with compatibility with Tevm as well

**Comment by tmm** [comment, marked as answer]
PR welcome!

**Reply by conwayconstar** [reply]
https://github.com/wevm/viem/pull/2242
